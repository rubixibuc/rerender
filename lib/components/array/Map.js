import React from "react";
import PropTypes from "prop-types";

const render = ({ items, children }) => (items || null) && items.map(children);

render.propTypes = {
  items: PropTypes.array,
  children: PropTypes.func.isRequired
};

export default render;
