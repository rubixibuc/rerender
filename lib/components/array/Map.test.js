import React from "react";
import Map from "./Map";
import renderer from "react-test-renderer";

test("should render nothing when items is undefined", () => {
  const component = renderer.create(
    <Map items={undefined}>{items => <div>{items}</div>}</Map>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render nothing when items is null", () => {
  const component = renderer.create(
      <Map items={null}>{items => <div>{items}</div>}</Map>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render nothing when items is empty", () => {
  const component = renderer.create(
      <Map items={[]}>{items => <div>{items}</div>}</Map>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render child for each item in items", () => {
  const component = renderer.create(
      <Map items={[1, 2]}>{(items, index) => <div key={index}>{items}</div>}</Map>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
