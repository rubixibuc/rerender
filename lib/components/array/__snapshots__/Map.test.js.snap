// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render child for each item in items 1`] = `
Array [
  <div>
    1
  </div>,
  <div>
    2
  </div>,
]
`;

exports[`should render nothing when items is empty 1`] = `null`;

exports[`should render nothing when items is null 1`] = `null`;

exports[`should render nothing when items is undefined 1`] = `null`;
